tests/hypervisor/test_local.py

```diff
@@ -5,40 +5,114 @@
+        result = virtwho.run_service()
+        thread_berfore = result["thread"]
+        logger.info(
+            f"Succeeded to start virt-who service and the thread_num is {thread_berfore}"
+        )
 
-        :title: virt-who: local: test http option
-        :id: a66787b3-1a5f-4b6f-9c09-0873d6490de3
-        :caseimportance: High
-        :tags: tier2
-        :customerscenario: false
-        :upstream: no
-        :steps:
-            1.
-        :expectedresults:
-            1.
-        """
-        logger.info("Succeeded to run the 'test_http_option'")
+        # update libvirt config, and restart libvirtd service
+        libvirt_conf = "/etc/libvirt/libvirtd.conf"
+        option_enable("listen_tls", libvirt_conf, ssh_host)
+        option_enable("listen_tcp", libvirt_conf, ssh_host)
+        option_enable("auth_tcp", libvirt_conf, ssh_host)
+        option_enable("tcp_port", libvirt_conf, ssh_host)
+
+        run_service(ssh_host, "libvirtd", "restart")
+        _, output = run_service(ssh_host, "libvirtd", "status")
+        assert "is running" in output or "Active: active (running)" in output
+
+        # check virt-who thread_num is changed or not
+        thread_after = virtwho.thread_number()
+        assert thread_berfore == thread_after
+
+        # recovery libvirt config
+        option_disable("listen_tls", libvirt_conf, ssh_host)
+        option_disable("listen_tcp", libvirt_conf, ssh_host)
+        option_disable("auth_tcp", libvirt_conf, ssh_host)
+        option_disable("tcp_port", libvirt_conf, ssh_host)
+
+        run_service(ssh_host, "libvirtd", "restart")
+        _, output = run_service(ssh_host, "libvirtd", "status")
+        assert "is running" in output or "Active: active (running)" in output
+
+
+def option_enable(option, file, ssh_host):
+    """
+    Enable the option for the specfic file
+    :param option: the name of the option
+    :param file: the name of the file
+    :param ssh_host: the ssh host for running the command
+    :return:
+    """
+    cmd = f'sed -i "s|^#{option}|{option}|g" {file}'
+    ret, output = ssh_host.runcmd(cmd)
+    if ret == 0:
+        logger.info(f"Succeeded to enable option {option}")
+        return True
+    else:
+        logger.error(f"Failed to enable option {option}")
+        return False
+
+
+def option_disable(option, file, ssh_host):
+    """
+    Disable the option for the specfic file
+    :param option: the name of the option
+    :param file: the name of the file
+    :param ssh_host: the ssh host for running the command
+    :return: Succeed to run the command, return True; Else, return False
+    """
+    cmd = f'sed -i "s|^{option}|#{option}|g" {file}'
+    ret, output = ssh_host.runcmd(cmd)
+    if ret == 0:
+        logger.info(f"Succeeded to disable option {option}")
+        return True
+    else:
+        logger.error(f"Failed to disable option {option}")
+        return False
+
+
+def run_service(ssh_host, name, action):
```

Comment:
we have the `operate_service()` in the runner.py, please think about if we can directly use it.

Reply:
`operate_service()` is 'systemctl' related, but here we need `service` operation